hecat/processors/url_check.py

```diff
@@ -95,3 +108,60 @@ def check_urls(step):
         print('\n'.join(errors))
         if 'errors_are_fatal' in step['module_options'].keys() and step['module_options']['errors_are_fatal']:
             sys.exit(1)
+
+def should_escalate(url, failed_urls_file, escalation_limit):
+    # Load existing failed URLs or create an empty list
+    try:
+        with open(failed_urls_file, 'r') as file:
+            failed_urls = [line.strip() for line in file.readlines()]
+    except FileNotFoundError:
+        return False
+
+    # Check if the URL has reached the escalation limit
+    if url in failed_urls:
+        index = failed_urls.index(url)
+        count = int(failed_urls[index + 1])
+        return count >= escalation_limit
+
+    return False
```

Comment:
There is some duplication here, I don't think this function is needed.
`handle_failed_url()` should contain all the logic to:
- check if the failed_urls_file exists, if not, create it
- check if the url is already in the file, if yes increment its counter, else append it to the file and set its counter to 1
- write the file to disk
- check if the counter is < escalation_limit, if yes log a warning, else log an error and append the error messaeg to `errors` list

Reply:
You were absolutely right. I created the function only to detect if an escalation would occur to log an error, which doesn't make any sense since I didn't do that either in the try-catch block 🤪. Since this if statement is not needed in the first place, I removed it together with the function in https://github.com/nodiscc/hecat/pull/133/commits/7c86db2d0b42e79fb39a7d4cd463a02077cfe123 & https://github.com/nodiscc/hecat/pull/133/commits/ffe7d5b8da819b737c0effb0a13f52d840477d3f (accidentally commit only half of the changes).